### PR TITLE
Improving token_auth_enforced description text in config.sample.php

### PR DESCRIPTION
**Pull request by mmattel**
## Description
Description improvement for the oAuth2 key ``token_auth_enforced``

## Motivation and Context
Improving the description to make it more understandable what it is about

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

The documentation will be updated asap after merging.

### CODE REVIEW

**Comment by mmattel** [comment]
Jenkins suicided ...

File: config/config.sample.php [hunk 1]
```diff
@@ -214,9 +214,11 @@
 'session_keepalive' => true,
 
 /**
- * Enforce token authentication for clients, which blocks requests using the user
- * password for enhanced security. Users need to generate tokens in personal settings
- * which can be used as passwords on their clients.
+ * Enforces oAuth2 token only authentication for clients connecting to ownCloud.
```
[the remark this answers is not in the document]
Comment (mmattel) [reply config/config.sample.php]: I am happy to further imrove this config.sample.php text.
Please do any suggestion and I will incorporate it.

**Comment by mmattel** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by ownclouders** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Comment by mmattel** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by ownclouders** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Comment by mmattel** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by ownclouders** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Review by SamuAlfageme — CHANGES_REQUESTED** [review]
It actually does not require the OAuth2 app to be installed/enabled - this setting also works out for application password logins (also token-based auth.). 

More info on https://github.com/owncloud/core/issues/29300#issuecomment-337935504

**Comment by mmattel** [comment]
@SamuAlfageme your statement is valid. 
I also see that the term "App Password" used in the personal settings page (just checked in my prod 10.0.7) has room for improvement to reflect what you say because we use token an password in the same context. 

I will remove the term OAuth2 from the first sentence and remove the whole last sentence as suggested.

Question: which role does OAuth2 play in this game?

**Comment by mmattel** [comment]
@SamuAlfageme I adopted the text, pls check

**Comment by phil-davis** [comment]
``oauth2`` is another token-based authentication method. So when ``token_auth_enforced`` is set to ``true`` then both "app/device password" and ``oauth2`` still work for authentication. "username-password" does not work (i.e. works only for the webUI login so you can get to the settings page to generate app passwords...)

**Comment by mmattel** [comment]
@phil-davis anything to improve on the text or are you ok with?

**Comment by phil-davis** [comment]
I am OK with it. I can't think how to mention ``oauth2`` without adding confusion.

**Comment by mmattel** [comment]
I have added you as approver 😃 
In case, please also approve the backport

**Review by phil-davis — APPROVED** [review]

**Comment by SamuAlfageme** [comment]
@mmattel @phil-davis 

> I can't think how to mention`oauth2` without adding confusion. 

... maybe consider to name _"the option to also use alternative authorization mechanisms like ownCloud's OAuth2 app"_ - also linking to https://marketplace.owncloud.com/apps/oauth2

**Review by SamuAlfageme — APPROVED** [review]

**Pull request merged**